desc/equilibrium/equilibrium.py

```diff
@@ -216,17 +224,33 @@ def __init__(
             ValueError,
             f"sym should be one of True, False, None, got {sym}",
         )
-        self._sym = setdefault(sym, getattr(surface, "sym", False))
+        if surface is not None:
+            self._sym = setdefault(sym, getattr(surface, "sym", False))
+        elif xsection is not None:
+            self._sym = setdefault(sym, getattr(xsection, "sym", False))
+        else:
+            self._sym = setdefault(sym, False)
         self._R_sym = "cos" if self.sym else False
         self._Z_sym = "sin" if self.sym else False
 
-        # surface
-        self._surface, self._bdry_mode = parse_surface(
-            surface, self.NFP, self.sym, self.spectral_indexing
+        errorif(
+            surface is not None and xsection is not None,
+            ValueError,
+            "Cannot specify both surface and xsection",
         )
-
-        # magnetic axis
-        self._axis = parse_axis(axis, self.NFP, self.sym, self.surface)
+        # Parse surface, magnetic axis and cross-section
```

Comment:
I feel like this is still a bit clunky and could lead to some weird inconsistencies. I think what we want is for at the end of this block, once surface, xsection, and axis are assigned, they should all be self-consistent (eg evaluating the xsection at rho=0 should be the same as the axis, evaluating the surface at zeta=0 should be the same as the xsection at rho=1 etc)

Right now I don't think that's the case, because the one thing (surface/xsection) that isn't given by the user is just set to the default for that class which generally won't match with the other object supplied by the user.

If we ensure that they're all self consistent here then we don't need to do anything with `set_initial_guess` since it will all be the same.

Here's a sketch of what I'm thinking:
```python
errorif(surface is not None and xsection is not None)
errorif(xsection is not None and axis is not None) # axis is redundant with xsection right?
if xsection is None and surface is None:
    self.surface = default_surface
    self.xsection = default_xsection
    # maybe need some extra logic if surface and xsection are None but axis is given, ie from NAE?
if xsection is not None:
    self.xsection = parse_xsection(xsection)
    self.surface = get_surface_from_section(self.xsection)
    self.axis = get_axis_from_xsection(self.xsection)
if surface is not None:
    self.surface = parse_surface(surface)
    if axis is None:
        self.axis = get_axis_from_surface(self.surface) # basically getting centroid of surface, M=0 modes
    else:
        self.axis = self.parse_axis(axis)
    self.xsection = get_xsection_from_surface_and_axis(self.surface, self.axis)
```

Reply:
I can update parse_surface function to use xsection too. I think the rest is already implemented. What you mean by get_xsection_from_surface would be basically rewriting the initial_guess stuff. Since we already do that at the end, I didn't spend time reimplementing it.